src/VersionParser.php

```diff
@@ -21,16 +24,22 @@ class VersionParser
     private $versions;
 
     /**
+     * @var Composer\VersionParser
+     */
+    private $parser;
+
+    /**
      * @var string
      */
-    private $modifier = '[._-]?(?:(stable|beta|b|RC|alpha|a|patch|pl|p)(?:[.-]?(\d+))?)?([.-]?dev)?';
+    const GIT_DATA_MATCH = '/.*(-\d+-g[[:alnum:]]{7})$/';
 
     /**
      * @param array $versions
      */
     public function __construct(array $versions = array())
     {
         $this->versions = $versions;
+        $this->parser = new Parser; 
```

Comment:
I remember the first time I saw this I was confused as I didn't know it was possible, and when you call a function `foo` which has no arguments you call `foo()`, not `foo`. So I did wonder why the seven hells people took the liberty to change it with `new` :trollface: 

In any case, it's my own bias as well and I don't think anyone would feel strongly about it 😄